src/models/DataSetStore.js

```diff
@@ -126,12 +141,61 @@ export default class DataSetStore {
         this.updateFromAssociations(fieldPath, oldValue);
     }
 
+    saveDataset() {
+        return new Promise(async (complete, error) => {
+            // maintenance-app uses a custom function to save the dataset as it needs to do
+            // some processing  not allowed by dataset.save(). The code in this method is copied
+            // from maintenance-app/src/EditModel/objectActions.js
```

Comment:
The most relevant part is how dataSetElements is saved, it need the dataset ID to be specified: `dataSet: { ...dataSet, id: dataSet.id || dataSetPayload.id },`. So it gets the UID from the server before saving, assigns the datSetElements (thus they are correctly set as "no override", and then the record is saved.